### PR TITLE
Rework ProbeDispatcher and ProbeWorker

### PR DESCRIPTION
**Pull request by kevinreniers**
I'm working on removing some dead code, code that doesn't _really_ do anything, reducing the data footprint between dispatcher and workers and adding more logging throughout.

### CODE REVIEW

File: src/Command/ProbeWorkerCommand.php [hunk 1]
```diff
@@ -88,219 +75,125 @@ protected function configure(): void
...
+        $type = $data['type'];
+        $this->logger->info(sprintf('worker %d received %s job from dispatcher', getmypid(), $type));
 
-        $command = null;
         try {
-            $command = $this->commandFactory->make($data['type'], $data);
+            $command = $this->commandFactory->make($type, $data);
+            $this->logger->info(sprintf('worker %d %s command initialized', getmypid(), $type));
         } catch (Exception $e) {
-            $this->sendResponse(
```
Comment (jimmycleuren) [inline src/Command/ProbeWorkerCommand.php:174]: Dispatcher will not be notified now ?

File: src/Command/ProbeWorkerCommand.php [hunk 2]
```diff
@@ -88,219 +75,125 @@ protected function configure(): void
...
-                            //'request' => $data,
-                            'pid' => getmypid(),
-                        ],
-                    ]);
+                    ];
+                    $this->sendResponse('probe', 200, $contents);
                     break;
+
                 default:
-                    $this->sendResponse(
```
Comment (jimmycleuren) [inline src/Command/ProbeWorkerCommand.php:260]: Dispatcher will not be notified now ?

File: src/Command/ProbeWorkerCommand.php [hunk 3]
```diff
@@ -88,219 +75,125 @@ protected function configure(): void
...
-                                'runtime' => time() - $startOfWork,
-                                'request' => $data,
-                                'pid' => getmypid(),
-                            ],
-                        ]
-                    );
+                    $this->logger->error("no handler defined for $type");
             }
         } catch (Exception $e) {
-            $this->sendResponse(
```
Comment (jimmycleuren) [inline src/Command/ProbeWorkerCommand.php:277]: Dispatcher will not be notified now ?

File: src/Command/ProbeDispatcherCommand.php [hunk 4]
```diff
@@ -284,95 +280,66 @@ public function sendInstruction(array $instruction, $expectedRuntime = null): vo
...
-        $pid = $debug['pid'];
-        $runtime = $debug['runtime'];
+        /** @var array|string $contents */
+        $contents = $response['contents'];
 
-        $this->logger->info("COMMUNICATION_FLOW: Master received $type response from worker $pid with a runtime of $runtime.");
+        $this->logger->info("dispatcher: $type response ($bytes bytes) received from worker $pid");
 
         switch ($type) {
-            case 'exception':
```
Comment (jimmycleuren) [inline src/Command/ProbeDispatcherCommand.php:318]: We should still do this ?

File: src/Command/ProbeDispatcherCommand.php [hunk 5]
```diff
@@ -284,95 +280,66 @@ public function sendInstruction(array $instruction, $expectedRuntime = null): vo
...
                     foreach ($this->configuration->getProbes() as $probe) {
                         $count += ceil($this->configuration->getProbeDeviceCount($probe->getId()) / $this->devicesPerWorker);
                     }
                     $this->workerManager->setNumberOfProbeProcesses(intval($count));
                 } else {
-                    $this->logger->info("Response ($status) from worker $pid received");
-                }
-                break;
-
-            case PostStatsHttpWorkerCommand::class:
```
Comment (jimmycleuren) [inline src/Command/ProbeDispatcherCommand.php:363]: This is nice to know ?

File: src/Command/ProbeWorkerCommand.php [hunk 6]
```diff
@@ -88,219 +75,125 @@ protected function configure(): void
...
-                ]
-            );
-
-            return;
-        }
+        $startedAt = time();
+        $this->logger->info(sprintf('worker %d has begun processing at %d', getmypid(), $startedAt));
 
-        if (!isset($data['type'])) {
-            $this->sendResponse(
```
Comment (jimmycleuren) [inline src/Command/ProbeWorkerCommand.php:148]: Dispatcher will not be notified now ?

File: src/Command/ProbeDispatcherCommand.php (hunk 4 above)
Comment (kevinreniers) [reply src/Command/ProbeDispatcherCommand.php:318]: Workers already log these themselves. It just leads to logging the same thing twice.

File: src/Command/ProbeDispatcherCommand.php (hunk 5 above)
Comment (kevinreniers) [reply src/Command/ProbeDispatcherCommand.php:363]: Should move this logging to the PostStatsHttpWorker instead. I can add some extra logging there?

Comment (jimmycleuren) [reply src/Command/ProbeDispatcherCommand.php:363]: Sounds good

File: src/Command/ProbeDispatcherCommand.php (hunk 4 above)
Comment (jimmycleuren) [reply src/Command/ProbeDispatcherCommand.php:318]: Ok if an empty response is sent

**Pull request merged**